include/slang/syntax/SyntaxNode.h

```diff
@@ -38,6 +38,10 @@ struct TokenOrSyntaxBase : public std::variant<Token, TNode> {
     /// if it's not actually a token).
     Token token() const { return std::get<0>(*this); }
 
+    /// Gets access to the object as a syntax node (throwing an exception
```

Comment:
I don't think you need this node() overload, it has the same return type as the const version so it's not really doing anything.

Reply:
I want to modify inplace.
``` cpp
class TestRewriter : public SyntaxRewriter<TestRewriter> {
    void handle(TypedefDeclarationSyntax& decl) {
    };
};
auto clone_obj = obj->clone(alloc);
clone_obj->visit(TestRewriter {});
```
Icannot archive without overload node function due to constness

Reply:
But they both return TNode, so how does this help? The `const` version will be called by even on non-const objects if there isn't this other overload, since it's always safe to call a const method on a non-const object.

Reply:
You are right, my visual C++ have success compile it. maybe my environment have some flawed.
It compiles ok on VC, GCC, clang